src/analyser.cpp

```diff
@@ -1117,12 +1240,785 @@ void Analyser::AnalyserImpl::analyseEquationAst(const AnalyserEquationAstPtr &as
 
     // Recursively check the given AST's children.
 
-    if (ast->mPimpl->mOwnedLeftChild != nullptr) {
-        analyseEquationAst(ast->mPimpl->mOwnedLeftChild);
+    analyseEquationAst(ast->mPimpl->mOwnedLeftChild);
+    analyseEquationAst(ast->mPimpl->mOwnedRightChild);
+}
+
+void Analyser::AnalyserImpl::updateUnitsMap(const ModelPtr &model,
+                                            const std::string &unitsName,
```

Comment:
Don't we have a very similar method elsewhere?

Reply:
Not as far as I am aware. Why? Which one do you have in mind?

Reply:
units.cpp:547

Reply:
Indeed, there is quite a bit of overlap. However, the code in `units.cpp` doesn't take into account the multiplier and (obviously) also doesn't take into what we did for the `UnitsMap` that we use for displaying issue messages.

So, at this stage (aka: time of the night), I feel like it's going to be a bit difficult to reuse that code. This being said, I have refatored some of the code in the analyser (see commit 14518461) and feel like, overall, it's a bit better than the one in `units.cpp`.

Reply:
I was wondering if one was the superset of the other. If so then maybe the code has a pathway to being reused.